WPMedia/fusion-news-theme-blocks
# THEMES-1039: Error handling for fetch content sources

## Description

This PR integrates the `handleFetchError` utility to the various `axios` calls within our content sources.

## Jira Ticket

- [THEMES-1039](https://arcpublishing.atlassian.net/browse/THEMES-1039)

## Acceptance Criteria

- GIVEN I am a PB Admin, WHEN I misconfigure a block using the above content sources, THEN: 
  - Confirm all content sources using `axios` fetch and apply this change to all 
  - The content source error is “Internal Server Error” 
  - The full `axios` objects are blocked from being exposed in the browser
  - Update the templates for the generator 

## Test Steps

1. Checkout this branch `git checkout THEMES-1039`
2. Checkout this `arc-themes-components` branch `git checkout THEMES-1039`
3. Run fusion repo with `npx fusion start`
4. Verify that the changed content sources are still returning as expected.
5. Verify that a `.catch()` has been added to every `axios` call in our blocks.

## Dependencies or Side Effects

- `arc-themes-components`: https://github.com/WPMedia/arc-themes-components/pull/172

## Author Checklist

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-1039]: https://arcpublishing.atlassian.net/browse/THEMES-1039?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
